lib/glob-assets.js

```diff
@@ -0,0 +1,45 @@
+const {basename} = require('path');
+const {isPlainObject, castArray, uniqWith} = require('lodash');
+const pReduce = require('p-reduce');
+const globby = require('globby');
+
+module.exports = async assets =>
+  uniqWith(
+    (await pReduce(
+      assets,
+      async (result, asset) => {
+        // Wrap single glob definition in Array
+        const glob = castArray(isPlainObject(asset) ? asset.path : asset);
+        // Skip solo negated pattern (avoid to include every non js file with `!**/*.js`)
+        if (glob.length > 1 || !glob[0].startsWith('!')) {
```

Comment:
doesn’t globby support negated patterns?

Reply:
It does and use it. This line is meant to prevent potential configuration mistakes:
`assets: [['dist/**', '!**/*.css'], ['package.json']]` => Get all files in dist except the css ones + `package.json`
`assets: [['!**/*.css'], ['package.json']]` => Get every files in the repo that are not css + `package.json`
The second one is problably a mistake.

So this like of code just skip the cases like `['!**/*.css']` as I don't think there is any situation in which you want to upload to Github "every files but some".

Reply:
ah got it. I’m sure someone will come up with a use case for that one day :D but yeah I see your point. Maybe we could log a comment for that?

Reply:
ok

Reply:
Done